cobc/typeck.c

```diff
@@ -4316,10 +4294,10 @@ cb_validate_program_environment (struct cb_program *prog)
 	}
 
 	/* Resolve the program collating sequences */
-	if (cb_validate_collating (prog->collating_sequence)) {
+	if (cb_validate_collating (prog, prog->collating_sequence)) {
 		prog->collating_sequence = NULL;
 	};
-	if (cb_validate_collating (prog->collating_sequence_n)) {
+	if (cb_validate_collating (prog, prog->collating_sequence_n)) {
```

Comment:
Possibly, at least in the case of a custom alphabet. We should probably adjust this function so that it considers whether we're dealing with a national collating sequence or not, and possibly have national variants of low_value / high_value.